### PR TITLE
Add changes from 1.17.4 to 1.17.6 too

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

New version of https://github.com/conda-forge/coin-or-clp-feedstock/pull/7 to incorporate all changes.

### CODE REVIEW

**Comment by BastianZim** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
     url: https://github.com/coin-or/Clp/archive/releases/{{ version }}.tar.gz
-    sha256: ef412cde00cb1313d9041115a700d8d59d4b8b8b5e4dde43e9deb5108fcfbea8
+    sha256: afff465b1620cfcbb7b7c17b5d331d412039650ff471c4160c7eb24ae01284c9
     patches:
        - 0001-Patch-for-downstream.patch
        - 0001-Add-second-downstream-patch.patch
 
 build:
-  number: 1
+  number: 3
```
Comment (BastianZim) [inline recipe/meta.yaml:16]: Last 1.17.6 build was 2: https://github.com/conda-forge/coin-or-clp-feedstock/commit/c8485e4706d9ade5fa8ff04553c37dd6b73f5859#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9a

**Comment by tkralphs** [comment]
I can't say that I understand the logic of conda versioning. So everything just gets committed to master in any order? How does conda figure out in the end what versions are available without any tags, branches, etc.? Anyway, as far as I can see, this looks fine, but I have to admit that I'm not enough of an expert to assess this. I can merge it if needed.

**Comment by BastianZim** [comment]
(This is for conda-forge) 
In short: Every time a commit is made to the feedstock, the CI builds a package based on what's in master, uploads it and uses the _version_ specified in the `meta.yaml` to make a unique combination of _name_ and _version_. If the _version_ already exists, it doesn't do anything.
Since sometimes a package has to be rebuilt (Incorrect previous build, new pin, new platform etc., new tests) without the _version_ being updated, conda-build also adds the _build number_ (Identified in the `meta.yaml`) to the name to distinguish packages, since it wouldn't be uploaded otherwise.
This results in the package being named _coin-or-clp-1.17.4-hc022024_1_ (_Version 1.17.4_ and _build 1_), for example. 
The versioning is therefore only based on the _version_ and _build number_ named in the`meta.yaml`.
This means that, in theory, it doesn't matter in which order you upload the packages as long as the combination of _version_ and _build number_ is unique.

Since this can cause some confusion conda-forge has established the practice of having different branches, one for each _version_. This allows you to get a better overview and keep things more organised.
One example here is the python feedstock (https://github.com/conda-forge/python-feedstock/branches) which has a branch for each minor version so when a patch gets released they are separated and you can still see where branch 3.8 is, even if you update 3.7.0 to 3.7.1.
This is purely for cosmetics though and generally only done when you know that you will need to add something to a lower version multiple times. Since I assume that here this is not the case and that this will be the only time that we need to build a new version of 1.17.4, https://github.com/conda-forge/coin-or-clp-feedstock/pull/5 was just merged into the normal master branch.

**Comment by BastianZim** [comment]
But this isn't super important, it's just to have the patch in all versions. If you're more comfortable with waiting for Wolf we can do that too.

**Comment by tkralphs** [comment]
OK, thanks for taking the time to explain that. I guessed it had to be something along those lines and I agree that in general, we won't need branches here. I'll just go ahead and merge.

**Pull request merged**